src/abacus-ts/lib/createStoreEffect.ts

```diff
@@ -13,8 +13,10 @@ export function createStoreEffect<T>(
   const removeStoreListener = store.subscribe(() => {
     const thisValue = selector(store.getState());
     if (thisValue !== lastValue) {
-      lastCleanup?.();
       lastValue = thisValue;
+      // NOTE: some cleanups dispatch actions which cause this to happen recursively.
+      // we must ensure that those actions don't change the state they subscribe to or this will go infinitely
+      lastCleanup?.();
```

Comment:
so a proper cleanup function for say fills shouldn't dispatch an action that changes the fills object?

Reply:
it can change fills but it can't change what the fills rest listens to, in this case just parent subaccount address and indexer url.

Reply:
Well in the case of fills:

- network id
- indexer readiness
- wallet address
- subaccount number